src/utils/queryWhoisServer.ts

```diff
@@ -0,0 +1,25 @@
+import net from "net";
+
+type GetWhoisData = (
+  domain: string,
+  whoisServer: string,
+  port: number
+) => Promise<string>;
+
+export const queryWhoisServer: GetWhoisData = async (
+  domain,
+  whoisServer,
+  port
+) => {
+  return new Promise((resolve, reject) => {
+    let data = "";
+    const socket = net.createConnection({ host: whoisServer, port: port }, () =>
+      socket.write(domain + "\r\n")
+    );
+    socket.setTimeout(60000);
```

Comment:
Also, it would be good to define local const. `const timeoutSeconds = 60` then

```suggestion
    socket.setTimeout(timeoutSeconds * 1000);
```